compiler/codegen/OMRCodeGenPhase.cpp

```diff
@@ -419,39 +419,43 @@ void
 OMR::CodeGenPhase::performSetupForInstructionSelectionPhase(TR::CodeGenerator * cg, TR::CodeGenPhase * phase)
    {
    TR::Compilation *comp = cg->comp();
-   // TODO (GuardedStorage)
-#if defined(OMR_GC_CONCURRENT_SCAVENGER)
-   traceMsg(comp, "GuardedStorage: in performSetupForInstructionSelectionPhase\n");
 
-   auto mapAllocator = getTypedAllocator<std::pair<TR::TreeTop*, TR::TreeTop*> >(comp->allocator());
-
-   std::map<TR::TreeTop*, TR::TreeTop*, std::less<TR::TreeTop*>, TR::typed_allocator<std::pair<TR::TreeTop*, TR::TreeTop*>, TR::Allocator> >
-      currentTreeTopToappendTreeTop(std::less<TR::TreeTop*> (), mapAllocator);
-
-   for (TR::PreorderNodeIterator iter(comp->getStartTree(), comp); iter != NULL; ++iter)
+   if (cg->isConcurrentScavengeEnabled())
```

Comment:
I can definitely sink this down into a z Systems specific codegen phase. Though given that Power also supports concurrent scavenge I suspect we want something like this for Power as well. I am questioning the existence of this code to begin with however. I think a better solution which I want to explore is to remove this code altogether (as noted in the comment near this snippet) and avoid rematerializing `TR::compressedRefs` trees into decompression sequences as this more closely models what our hardware instruction does (it loads and shifts all in one go so we shouldn't really model the load before the shift in the IL).

Given my explanation please recommend whether you would like this pushed into a z Systems specific codegen phase.